### PR TITLE
Support merge conv bn layers at testing.

### PR DESCRIPTION
**Pull request by yhcao6**
| Model             | Original Inf time (s/iter) | New Inf time (s/iter) |
|-------------------|----------------------------|-----------------------|
| Faster R-CNN R-50 | 16.0                       | 16.3                  |
| Mask R-CNN R-50   | 10.9                       | 11.1                  |
| Retinanet R-50    | 16                         | 16.3                  |

### CODE REVIEW

File: tools/merge_conv_bn.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    merged_conv_w = conv_w * factor.reshape([conv.out_channels, 1, 1, 1])
+    merged_conv_b = (conv_b - bn.running_mean) * factor + bn.bias
+
+    merged_conv = nn.Conv2d(
+        conv.in_channels,
+        conv.out_channels,
+        conv.kernel_size,
+        conv.stride,
+        conv.padding,
+        bias=True)
```
Comment (hellock) [inline tools/merge_conv_bn.py]: No need to construct a new conv layer, just set the weight and bias of the original conv.

File: tools/merge_conv_bn.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import argparse
+
+import torch
+import torch.nn as nn
+from mmcv.runner import save_checkpoint
+
+from mmdet.apis import init_detector
+
+
+def merge_conv_bn(conv, bn):
```
Comment (hellock) [inline tools/merge_conv_bn.py]: Add a docstring for better understandings for junior users.

File: tools/test.py [hunk 3]
```diff
@@ -49,6 +50,11 @@ def parse_args():
     parser.add_argument('config', help='test config file path')
     parser.add_argument('checkpoint', help='checkpoint file')
     parser.add_argument('--out', help='output result file in pickle format')
+    parser.add_argument(
+        '--merge_conv_bn',
```
Comment (hellock) [inline tools/test.py]: change all `merge` into `fuse`

**Comment by xmyqsh** [comment]
@yhcao6 
Any idea for merging conv bn relu?

**Comment by hellock** [comment]
> @yhcao6
> Any idea for merging conv bn relu?

No such implementation is available.

**Comment by xmyqsh** [comment]
You are right.
Someone refers to this several years ago. But no real repo released.

Finally, CUTLASS said it could merge element-wise operators such as relu into GEMM.
I think this is a valuable technic because RegNet proves that the activation layer is the bottleneck of cnn on GPU.
